lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1720,15 +1721,18 @@ void GrandCentralPass::runOnOperation() {
   if (!effectiveDUT)
     effectiveDUT = cast<FModuleOp>(
         *instancePaths->instanceGraph.getTopLevelNode()->getModule());
-  auto dfRange =
-      llvm::depth_first(instancePaths->instanceGraph.lookup(effectiveDUT));
+  // Use filter graph but don't filter anything yet.
+  auto fGraph =
+      make_filtered_graph(instancePaths->instanceGraph.lookup(effectiveDUT),
+                          [](auto edge) { return true; });
+  auto dfRange = make_df_range(fGraph);
   for (auto i = dfRange.begin(), e = dfRange.end(); i != e;) {
-    auto module = cast<FModuleLike>(*i->getModule());
+    auto module = cast<FModuleLike>(*(**i)->getModule());
     if (AnnotationSet(module).hasAnnotation(companionAnnoClass)) {
       i.skipChildren();
       continue;
     }
```

Comment:
For completeness, does it make sense to also have node more integrated node filtering? What I didn't like a lot about this code was the need for the `i.skipChildren()` here. However, maybe that's fine?

Reply:
Totally, node filtering would be a good thing to support while at it and should be a short hop-skip given the rest -- although how this works re:iterating all nodes `node_begin/node_end` (not defined for the FilteredGraph at all presently) is less clear.
I was thinking if this was useful the NodeRef/EdgeRef maybe should point to the "FilteredGraph" itself which could contain the filters for each or whatever makes sense.  Bit more indirection, but also makes the comparison operators more natural, was the thinking.

Offhand I think a similar effect could be had by filtering out edges if the edge's destination (here, `getTarget()`) has the companion annotation?  WDYT?